airflow/providers/amazon/aws/operators/boto3.py

```diff
@@ -0,0 +1,92 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable, Sequence
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+from airflow.models import BaseOperator
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class Boto3BaseOperator(BaseOperator):
+    """Base boto3 operator for interating with AWS via boto3"""
+
+    template_fields: Sequence[str] = ("boto3_kwargs",)
+
+    def __init__(
+        self,
+        *args,
+        aws_conn_id: str = "aws_conn_id",
+        hook_params: dict = None,
+        boto3_callable: str = None,
```

Comment:
Just personal thoughts.

Might be better use two arguments instead of one:

```python
def __init__(
    self,
    client_type: str,
    client_method: str,
    ...
)
```

In this case
1. We do not need split method
2. It might be templated fields
3. In `execute` we might also have addition validation:
   + Is this service available: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.get_available_services
   + Is `client_method` valid identifier: https://docs.python.org/3.7/library/stdtypes.html#str.isidentifier

Reply:
I agree with this. I also prefer having two explicit arguments instead of one

Reply:
1 & 2 done, for 3 I'd defer that validation to `AwsBaseHook`

Reply:
Why deferring it to `AwsBaseHook`? To me it makes more sense to add that validation in here. Plus, in `AwsBaseHook` you dont have the `client_method` information. Thus, you wont be able to validate this information

Reply:
I mean checking if `client` available should be done in the `AwsBaseHook`, since it creates the client and works with `boto3` directly. This way we have that validation for all `AwsBaseHook` consumers.

Of course it makes a sense to check if `client` has a `method` in the `Boto3Operator`. This validation should happen after the `client` was successfully created. Implemented [here](https://github.com/apache/airflow/pull/27656/commits/3982d90fc36ea713f9e542737bfde00e66e17cca#diff-82af9cd67b3a6cb52cbb96af5e56d886f83a03600f6e54be4e062900479c8b91R73)